prov/efa/configure.m4

```diff
@@ -99,6 +99,13 @@ AC_DEFUN([FI_EFA_CONFIGURE],[
 
 	AS_IF([test "$enable_efa" = "no"], [efa_happy=0])
 
+	# TODO: fix this once windows is supported
+	AS_IF([test $linux -ne 1 || { test x$host_cpu != xx86_64 && test x$host_cpu != xaarch64 ;}],
```

Comment:
config.guess does not always canonicalize to x86_64.  It will sometimes canonicalize to amd64 and some others.  Some will even report i686 and still be a 64 bit platform.  Life was a mess.  The most portable way of doing this test would be:

```
AS_IF([test $linux -ne 1],
    [AC_MSG_WARN[(EFA provider only supports Linux.  Cannot build.])
      efa_happy=0],
     [case "${host}" in
        x86_64-*x32|i?86-*|x86_64*|amd64*)
            AS_IF([test "$ac_cv_sizeof_long" = "4"], [AC_MSG_WARN....])
     esac])
 
case "${host}" in
        x86_64-*x32|i?86-*|x86_64*|amd64*)
            if test "$ac_cv_sizeof_long" = "4" ; then

Reply:
Thanks for the review, I had a feeling that this was wrong.

I've been looking at the compiler warnings and I think they can be fixed. As Sean mentioned most are print format specifier issues, but there are some casting issues as well that need to be resolved.

I wonder if we should disable 32 bit builds anyway since we don't support it anywhere.

Reply:
Disabling 32 bit builds makes sense, given we never test it and don't have a use case.  And is way simpler than what you're trying to do.  Just look at the size of void* and disable if it is 4.  I assume ac_cv_sizeof_void_p is already set, but you may have to AC_CHECK_SIZEOF(void*) to make sure it is set.

Reply:
I like it, thanks. :) See PR #7221